telegram/ext/_extbot.py

```diff
@@ -750,7 +750,9 @@ async def stop_poll(
         return await super().stop_poll(
             chat_id=chat_id,
             message_id=message_id,
-            reply_markup=self._replace_keyboard(reply_markup),
+            reply_markup=cast(
+                Optional["InlineKeyboardMarkup"], self._replace_keyboard(reply_markup)
+            ),
```

Comment:
maybe we can instead improve the typing of `_replace_keyboard` here via a typevar, i.e.

```python
_replace_keyboard(self, reply_markup: Optional[KT]) -> Optional[KT]:
```